File: js/events.ts
import { AddRectangle } from "./generated/outgoing_events/add_rectangle"

export type ColorEvent = {
	event_type: "color"
	params: String
}
export type AddRectangleEvent = {
	event_type: "AddRectangle"
	params: AddRectangle
}
export type EditRectangleEvent = {
	event_type: "EditRectangle"
	params: AddRectangle
}


export type Event = ColorEvent | AddRectangleEvent | EditRectangleEvent

File: event_types/src/events.rs
pub(crate)mod params;
use serde::{Deserialize,Serialize};
use params::AddRectangle;
use schemars::JsonSchema;
#[derive(Deserialize, Serialize, JsonSchema)]
#[serde(deny_unknown_fields)]
///Every event that the editor sends to the game
pub enum Event {
    Color(String),
    AddRectangle(AddRectangle),
    EditRectangle(AddRectangle),
}
#[derive(Deserialize, Serialize, JsonSchema)]
///Every event that the game sends to the editor
#[serde(deny_unknown_fields)]
pub enum SendEvents {
    EditRectangle(AddRectangle),
}
